cargo/cargo_build_script.bzl

```
@@ -52,6 +52,8 @@ def _cargo_build_script_run(ctx, script):
     # Pull in env vars which may be required for the cc_toolchain to work (e.g. on OSX, the SDK version).
     # We hope that the linker env is sufficient for the whole cc_toolchain.
     cc_toolchain, feature_configuration = get_cc_toolchain(ctx)
+    cc_env = get_cc_compile_env(cc_toolchain, feature_configuration)
+    env.update(**cc_env)
```

I am not sure either this is the best solution but I think we are better with it than without.

```suggestion
    # TODO: Trim down the list of environment shipped.
    env.update(**cc_env)
```

---

If you'd prefer to keep env changes to a minimum, it seems as if INCLUDE is sufficient for the CC-using crates in my project to compile:

```diff
diff --git a/cargo/cargo_build_script.bzl b/cargo/cargo_build_script.bzl
index 0725c4c..a13826b 100644
--- a/cargo/cargo_build_script.bzl
+++ b/cargo/cargo_build_script.bzl
@@ -53,7 +53,9 @@ def _cargo_build_script_run(ctx, script):
     # We hope that the linker env is sufficient for the whole cc_toolchain.
     cc_toolchain, feature_configuration = get_cc_toolchain(ctx)
     cc_env = get_cc_compile_env(cc_toolchain, feature_configuration)
-    env.update(**cc_env)
+    includes = cc_env.get("INCLUDE")
+    if includes:
+        env["INCLUDE"] = includes
     _, _, linker_env = get_linker_and_args(ctx, cc_toolchain, feature_configuration, None)
     env.update(**linker_env)
```

Would you like me to push this change?

---

I've tentatively made this change and rebased on master.